libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -291,6 +271,76 @@ export class Simulator {
     }
   }
 
+  private async startResources() {
+    // create an execution graph and start all resources in topological order
+    const execution = this._model.graph.toposort();
+
+    for (const wave of execution) {
+      if (this.concurrency) {
+        await Promise.all(wave.map((r) => this.startResource(r)));
+      } else {
+        for (const r of wave) {
+          await this.startResource(r);
+        }
+      }
+    }
+  }
+
+  /**
+   * Updates the running simulation with a new version of the app. This will create/update/delete
+   * resources as necessary to get to the desired state.
+   * @param simDir The path to the new version of the app
+   */
+  public async update(simDir: string) {
```

Comment:
Maybe this can be saved for a future optimization, but I feel like both `stop` and `update` can use similar logic as `start`. i.e. utilize topological order and concurrency.
In the case of `stop`, it'd be a reverse toposort

Reply:
Both "update" and "delete" are using topological order implicitly.

Update calls startResources which starts in order.

Stop first stops it's dependents which implies it doesn't in reverse topo order

Reply:
True, the order is correct so I moreso mean the concurrency aspect, which only startResources is making use of

Reply:
I'll add a note

Reply:
I decided to get rid of the concurrency in the PR (both for "start" and "stop"). I'll leave it for a future optimization.

So basically, removed the `toposort` method and in `startResource`, I am first making sure all dependencies are started (simply recursive DFS).